module2.js

```diff
@@ -51,8 +62,13 @@ function question2(carObj) {
 */
 
 function question3(dogs) {
-  // TODO
-  return /* ??? */
+  let result = []
+  dogs.map(dog => {
+    if (dog.weeksOld > 3) {
+      result.push(dog.name)
+    }
+  })
```

Comment:
This is actually more efficient than the implementation the description suggests, as it only loops over the array once; good job!